spacy/training/loggers.py

```diff
@@ -137,3 +137,77 @@ def finalize() -> None:
         return log_step, finalize
 
     return setup_logger
+
+
+@registry.loggers("spacy.CometLogger.v1")
+def comet_logger(project_name: str, remove_config_values: List[str] = []):
+    try:
+        import comet_ml
+    except ImportError:
+        print("You need to install `comet_ml` to use the CometLogger: see: https://comet.ml/docs/python-sdk/spacy/")
+
+    console = console_logger(progress_bar=False)
+
+    def setup_logger(
+        nlp: "Language", stdout: IO = sys.stdout, stderr: IO = sys.stderr
+    ) -> Tuple[Callable[[Dict[str, Any]], None], Callable[[], None]]:
+
+        try:
+            experiment = comet_ml.Experiment(project_name=project_name)
+        except Exception:
+            experiment = None
+            print("You need to configure Comet, and run appropriately: see: https://comet.ml/docs/python-sdk/spacy/")
+
+        # Get the config, removing items if necessary:
+        config = nlp.config.interpolate()
+        config_dot = util.dict_to_dot(config)
+        for field in remove_config_values:
+            del config_dot[field]
+        config = util.dot_to_dict(config_dot)
+
+        if experiment is not None:
+            experiment.log_asset_data(config, "spacy-config.cfg")
+
+        # Get methods for step console processing:
+        console_log_step, console_finalize = console(nlp, stdout, stderr)
+
+        def flatten_dictionary(prefix, dictionary, results):
+            """
+            Recursively construct key/value pair.
+            """
+            for key, value in dictionary.items():
+                if isinstance(value, dict):
+                    if prefix:
+                        flatten_dictionary(prefix + "/" + key, value, results)
+                    else:
+                        flatten_dictionary(key, value, results)
+                else:
+                    if prefix:
+                        results[prefix + "/" + key] = value
+                    else:
+                        results[key] = value
```

Comment:
Again it feels like this should probably be a helper/utility function

Reply:
Moved to utils